lib/s3.py

```diff
@@ -22,28 +22,27 @@
 ACL_PUBLIC = 'x-amz-acl:public-read'
 
 
-def split_bucket(url):
-    return url.hostname.split('.', 1)
+def get_key(hostname):
+    s3_key_dir = xdg_config_home('cockpit-dev/s3-keys', envvar='COCKPIT_S3_KEY_DIR')
 
+    # ie: 'cockpit-images.eu.linode.com' then 'eu.linode.com', then 'linode.com'
+    while '.' in hostname:
+        try:
+            with open(os.path.join(s3_key_dir, hostname)) as fp:
+                return fp.read().split()
+        except FileNotFoundError:
+            _, _, hostname = hostname.partition('.')  # strip a leading component
 
-def get_key_filename(endpoint):
-    s3_key_dir = xdg_config_home('cockpit-dev/s3-keys', envvar='COCKPIT_S3_KEY_DIR')
-    return os.path.join(s3_key_dir, endpoint)
+    return None
 
 
 def is_key_present(url):
```

Comment:
This function is now redundant -- you can call get_key() directly and interpret the value as boolean.

Reply:
It's true, but that exposes the key on the API.  I'd prefer not to do that...

(I know there's no real protection here, and that this is not really an "API" in any meaningful sense, but still...)